[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# file-not-included

Disallow the use of Panda CSS in files that are not included in the specified Panda CSS `include` config.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/all`.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/recommended`.

## Rule details

âœ”ï¸ Examples of **correct** code for a file named `App.tsx`:
```js
// File App.tsx is covered in the include config, so it's okay to import css and Circle from panda into it.

import { css } from './panda/css';
import { Circle } from './panda/jsx';
```

âŒ Examples of **incorrect** code for a file named `Invalid.tsx`:
```js
// File Invalid.tsx is not covered in the include config, so imporing css and Circle from panda into it is not allowed.

import { css } from './panda/css';
import { Circle } from './panda/jsx';
```

## Resources

* [Rule source](/plugin/src/rules/file-not-included.ts)
* [Test source](/plugin/tests/file-not-included.test.ts)
