[//]: # 'This file is generated by eslint-docgen. Do not edit it directly.'

# file-not-included

Disallow the use of panda css in files that are not included in the specified panda `include` config.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/all`.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/recommended`.

## Rule details

âŒ Examples of **incorrect** code for a file named `invalid.tsx`:

```js
import { css } from './panda/css'
import { Circle } from './panda/jsx'
```

âœ”ï¸ Examples of **correct** code for a file named `valid.tsx`:

```js
import { css } from './panda/css'
import { Circle } from './panda/jsx'
```

## Resources

- [Rule source](/plugin/src/rules/file-not-included.ts)
- [Test source](/tests/file-not-included.test.ts)
