[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-config-function-in-source

Prohibit the use of config functions outside the Panda config.

📋 This rule is enabled in `plugin:@pandacss/all`.

📋 This rule is enabled in `plugin:@pandacss/recommended`.

🔧 The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## Rule details

❌ Examples of **incorrect** code for a file named `valid.tsx`:
```js
import { defineKeyframes } from '@pandacss/dev';

const keyframes = defineKeyframes({
  fadeIn: {
    '0%': { opacity: '0' },
    '100%': { opacity: '1' },
  },
})
```

✔️ Examples of **correct** code for a file named `panda.config.ts`:
```ts
import { defineKeyframes } from '@pandacss/dev';

const keyframes = defineKeyframes({
  fadeIn: {
    '0%': { opacity: '0' },
    '100%': { opacity: '1' },
  },
})
```

🔧 Examples of code **fixed** by this rule for a file named `valid.tsx`:
```js
import { defineKeyframes } from '@pandacss/dev'; /* → */ import { defineKeyframes } from '@pandacss/dev';
                                                 /* → */
const keyframes = defineKeyframes({              /* → */
  fadeIn: {                                      /* → */
    '0%': { opacity: '0' },                      /* → */
    '100%': { opacity: '1' },                    /* → */
  },                                             /* → */
})                                               /* → */
```

## Resources

* [Rule source](/plugin/src/rules/no-config-function-in-source.ts)
* [Test source](/test/no-config-function-in-source.test.ts)
