[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-debug

Disallow the inclusion of the debug attribute when shipping code to the production environment.

📋 This rule is enabled in `plugin:@pandacss/all`.

📋 This rule is enabled in `plugin:@pandacss/recommended`.

🔧 The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## Rule details

❌ Examples of **incorrect** code:
```js
import { css } from './panda/css';

const styles = css({ bg: 'gray.900', debug: true });

import { css } from './panda/css';

const styles = css.raw({ color: 'gray.50', debug: true })
```

✔️ Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ bg: 'gray.900' });

import { css } from './panda/css';

const styles = css.raw({ color: 'gray.50' })
```

🔧 Examples of code **fixed** by this rule:
```js
import { css } from './panda/css';                        /* → */ import { css } from './panda/css';
                                                          /* → */
const styles = css({ bg: 'gray.900', debug: true });      /* → */ const styles = css({ bg: 'gray.900', });

import { css } from './panda/css';                        /* → */ import { css } from './panda/css';
                                                          /* → */
const styles = css.raw({ color: 'gray.50', debug: true }) /* → */ const styles = css.raw({ color: 'gray.50', })
```

## Resources

* [Rule source](/plugin/src/rules/no-debug.ts)
* [Test source](/tests/no-debug.test.ts)
