[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-debug

Disallow the inclusion of the debug attribute when shipping code to the production environment.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/all`.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/recommended`.

## Rule details

âŒ Examples of **incorrect** code:
```js
import { css } from './panda/css';

const styles = css({ marginLeft: '4', debug: true });
```
```js

import { css } from './panda/css';

function App(){
  return <div className={css({ background: 'red.100', debug: true })} />;
};
```
```js

import { Circle } from './panda/jsx';

function App(){
  return <Circle _hover={{ position: 'absolute' }} debug />;
}
```

âœ”ï¸ Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ marginLeft: '4' });
```
```js

import { css } from './panda/css';

function App(){
  return <div className={css({ background: 'red.100' })} />;
};
```
```js

import { Circle } from './panda/jsx';

function App(){
  return <Circle _hover={{ position: 'absolute' }} />;
}
```

## Resources

* [Rule source](/plugin/src/rules/no-debug.ts)
* [Test source](/tests/no-debug.test.ts)
