[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-debug

Disallow the inclusion of the debug attribute when shipping code to the production environment.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/all`.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/recommended`.

ğŸ”§ The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## Rule details

âŒ Examples of **incorrect** code:
```js
import { css } from './panda/css';

const styles = css({ bg: 'gray.900', debug: true });
;

import { css } from './panda/css';

const styles = css.raw({ color: 'gray.50', debug: true });
```

âœ”ï¸ Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ bg: 'gray.900' });
;

import { css } from './panda/css';

const styles = css.raw({ color: 'gray.50' });
```

ğŸ”§ Examples of code **fixed** by this rule:
```js
import { css } from './panda/css';                         /* â†’ */ import { css } from './panda/css';
                                                           /* â†’ */
const styles = css({ bg: 'gray.900', debug: true });       /* â†’ */ const styles = css({ bg: 'gray.900', });
;                                                          /* â†’ */ ;

import { css } from './panda/css';                         /* â†’ */ import { css } from './panda/css';
                                                           /* â†’ */
const styles = css.raw({ color: 'gray.50', debug: true }); /* â†’ */ const styles = css.raw({ color: 'gray.50', });
```

## Resources

* [Rule source](/plugin/src/rules/no-debug.ts)
* [Test source](/tests/no-debug.test.ts)
