[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-debug

Disallow the inclusion of the debug attribute when shipping code to the production environment.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/all`.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/recommended`.

ğŸ”§ The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## Rule details

âŒ Examples of **incorrect** code for a file named `invalid.tsx`:
```js
import { css } from './panda/css';
import { styled, Circle } from './panda/jsx';

const styles = css({ bg: "red", debug: true });

import { css } from './panda/css';
import { styled, Circle } from './panda/jsx';

function App() {
  const PandaComp = styled(div);
  return <PandaComp css={{ debug: true }} />;
}
```

âœ”ï¸ Examples of **correct** code for a file named `valid.tsx`:
```js
import { css } from './panda/css';
import { styled, Circle } from './panda/jsx';

const randomFunc = f({ debug: true })
```

ğŸ”§ Examples of code **fixed** by this rule for a file named `invalid.tsx`:
```js
import { css } from './panda/css';              /* â†’ */ import { css } from './panda/css';
import { styled, Circle } from './panda/jsx';   /* â†’ */ import { styled, Circle } from './panda/jsx';
                                                /* â†’ */
const styles = css({ bg: "red", debug: true }); /* â†’ */ const styles = css({ bg: "red", });

import { css } from './panda/css';              /* â†’ */ import { css } from './panda/css';
import { styled, Circle } from './panda/jsx';   /* â†’ */ import { styled, Circle } from './panda/jsx';
                                                /* â†’ */
function App() {                                /* â†’ */ function App() {
  const PandaComp = styled(div);                /* â†’ */   const PandaComp = styled(div);
  return <PandaComp css={{ debug: true }} />;   /* â†’ */   return <PandaComp css={{}} />;
}                                               /* â†’ */ }
```

## Resources

* [Rule source](/plugin/src/rules/no-debug.ts)
* [Test source](/test/no-debug.test.ts)
