[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-dynamic-styling

Ensure user doesn't use dynamic styling at any point. Prefer to use static styles, leverage css variables or recipes for known dynamic styles.

📋 This rule is enabled in `plugin:@pandacss/all`.

📋 This rule is enabled in `plugin:@pandacss/recommended`.

## Rule details

❌ Examples of **incorrect** code:
```js
import { css } from './panda/css';

const styles = css({ bg: color });

import { Circle } from './panda/jsx';

function App(){
  const bool = true;
  return  <Circle debug={bool} />;
};

import { styled } from './panda/jsx';

function App(){
  const color = 'red.100';
  return  <styled.div color={color} />;
}
```

✔️ Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ bg: 'gray.900' });

import { Circle } from './panda/jsx';

function App(){
  return <Circle debug={true} />;
};

import { styled } from './panda/jsx';

function App(){
  return <styled.div color='red.100' />;
}
```

## Resources

* [Rule source](/plugin/src/rules/no-dynamic-styling.ts)
* [Test source](/tests/no-dynamic-styling.test.ts)
