[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-debug

Disallow the inclusion of the debug attribute when shipping code to the production environment.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/all`.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/recommended`.

## Rule details

âŒ Examples of **incorrect** code:
```js
import { css } from './panda/css';

const styles = css({ bg: 'gray.900', debug: true });
```
```js

import { css } from './panda/css';

const styles = css.raw({ color: 'gray.50', debug: true })
```

âœ”ï¸ Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ bg: 'gray.900' });
```
```js

import { css } from './panda/css';

const styles = css.raw({ color: 'gray.50' })
```

## Resources

* [Rule source](/plugin/src/rules/no-debug.ts)
* [Test source](/tests/no-debug.test.ts)
