[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-debug

Disallow the inclusion of the debug attribute when shipping code to the production environment.

📋 This rule is enabled in `plugin:@pandacss/all`.

📋 This rule is enabled in `plugin:@pandacss/recommended`.

🔧 The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## Rule details

❌ Examples of **incorrect** code for a file named `invalid.tsx`:
```js
import { css } from './panda/css';
import { styled, Circle } from './panda/jsx';

const styles = css({ bg: "red", debug: true });

import { css } from './panda/css';
import { styled, Circle } from './panda/jsx';

function App() {
  const PandaComp = styled(div);
  return <PandaComp css={{ debug: true }} />;
}
```

✔️ Examples of **correct** code for a file named `valid.tsx`:
```js
import { css } from './panda/css';
import { styled, Circle } from './panda/jsx';

const randomFunc = f({ debug: true })
```

🔧 Examples of code **fixed** by this rule for a file named `invalid.tsx`:
```js
import { css } from './panda/css';              /* → */ import { css } from './panda/css';
import { styled, Circle } from './panda/jsx';   /* → */ import { styled, Circle } from './panda/jsx';
                                                /* → */
const styles = css({ bg: "red", debug: true }); /* → */ const styles = css({ bg: "red", });

import { css } from './panda/css';              /* → */ import { css } from './panda/css';
import { styled, Circle } from './panda/jsx';   /* → */ import { styled, Circle } from './panda/jsx';
                                                /* → */
function App() {                                /* → */ function App() {
  const PandaComp = styled(div);                /* → */   const PandaComp = styled(div);
  return <PandaComp css={{ debug: true }} />;   /* → */   return <PandaComp css={{}} />;
}                                               /* → */ }
```

## Resources

* [Rule source](/plugin/src/rules/no-debug.ts)
* [Test source](/test/no-debug.test.ts)
