[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-dynamic-styling

Ensure user doesn't use dynamic styling at any point. Prefer to use static styles, leverage css variables or recipes for known dynamic styles.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/all`.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/recommended`.

## Rule details

âŒ Examples of **incorrect** code for a file named `invalid.tsx`:
```js
import { css } from './panda/css'
import { styled, Circle } from './panda/jsx'

const styles = css({ bg: color });

import { css } from './panda/css'
import { styled, Circle } from './panda/jsx'

<Circle debug={bool} />;

import { css } from './panda/css'
import { styled, Circle } from './panda/jsx'

<styled.div color={color} />
```

âœ”ï¸ Examples of **correct** code for a file named `valid.tsx`:
```js
import { css } from './panda/css'
import { styled, Circle } from './panda/jsx'

const styles = css({ bg: "red" });

import { css } from './panda/css'
import { styled, Circle } from './panda/jsx'

const styles = css({ bg: `red` });

import { css } from './panda/css'
import { styled, Circle } from './panda/jsx'

const styles = css({ bg: 1 });

import { css } from './panda/css'
import { styled, Circle } from './panda/jsx'

const styles = css({ debug: true });

import { css } from './panda/css'
import { styled, Circle } from './panda/jsx'

<Circle debug={true} />;

import { css } from './panda/css'
import { styled, Circle } from './panda/jsx'

<Circle color={"red"} />;

import { css } from './panda/css'
import { styled, Circle } from './panda/jsx'

<Circle color={`red`} />
```

## Resources

* [Rule source](/plugin/src/rules/no-dynamic-styling.ts)
* [Test source](/test/no-dynamic-styling.test.ts)
