[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-deprecated-tokens

Disallow the use of deprecated tokens within token function syntax.

📋 This rule is enabled in `plugin:@pandacss/all`.

📋 This rule is enabled in `plugin:@pandacss/recommended`.

## Rule details

❌ Examples of **incorrect** code:
```js
import { css } from './panda/css';

// Assumes that the token is deprecated
const styles = css({ color: 'red.400' })
```

✔️ Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ color: 'red.100' })
```

## Resources

* [Rule source](/plugin/src/rules/no-deprecated-tokens.ts)
* [Test source](/plugin/tests/no-deprecated-tokens.test.ts)
