[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-dynamic-styling

Ensure user doesn't use dynamic styling at any point. 
Prefer to use static styles, leverage css variables or recipes for known dynamic styles.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/all`.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/recommended`.

## Rule details

âŒ Examples of **incorrect** code:
```js
import { css } from './panda/css';

const color = 'red.100';
const styles = css({ bg: color });
```
```js

import { stack } from './panda/patterns';

const align = 'center';
const styles = stack({ align: align });
```
```js

import { Circle } from './panda/jsx';

function App(){
  const bool = true;
  return <Circle debug={bool} />;
};
```
```js

import { styled } from './panda/jsx';

function App(){
  const color = 'red.100';
  return <styled.div color={color} />;
};
```
```js

import { css } from './panda/css';

const property = 'background';
const styles = css({ [property]: 'red.100' });
```
```js

import { cva,sva } from './panda/css';

function App(){
  const computedValue = "value"
  const heading = cva({
    variants: {
      [computedValue]: {
        color: "red.100",
      }
    }
  });
}
```

âœ”ï¸ Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ bg: 'gray.900' });
```
```js

import { Circle } from './panda/jsx';

function App(){
  return <Circle debug={true} />;
};
```
```js

import { styled } from './panda/jsx';

function App(){
  return <styled.div color='red.100' />;
}
```

## Resources

* [Rule source](/plugin/src/rules/no-dynamic-styling.ts)
* [Test source](/tests/no-dynamic-styling.test.ts)
