[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-escape-hatch

Prohibit the use of escape hatch syntax in the code.

📋 This rule is enabled in `plugin:@pandacss/all`.

🔧 The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## Rule details

❌ Examples of **incorrect** code:
```js
import { css } from './panda/css';

const styles = css({ marginLeft: '[4px]' });

import { css } from './panda/css';

function App(){
  return  <div className={css({ background: '[#111]' })} />;
};

import { Circle } from './panda/jsx';

function App(){
  return  <Circle _hover={{ position: '[absolute]' }} />;
}
```

✔️ Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ marginLeft: '4' });

import { css } from './panda/css';

function App(){
  return  <div className={css({ background: 'red.100' })} />;
};

import { Circle } from './panda/jsx';

function App(){
  return  <Circle _hover={{ position: 'absolute' }} />;
}
```

🔧 Examples of code **fixed** by this rule:
```js
import { css } from './panda/css';                           /* → */ import { css } from './panda/css';
                                                             /* → */
const styles = css({ marginLeft: '[4px]' });                 /* → */ const styles = css({ marginLeft: '4px' });

import { css } from './panda/css';                           /* → */ import { css } from './panda/css';
                                                             /* → */
function App(){                                              /* → */ function App(){
  return  <div className={css({ background: '[#111]' })} />; /* → */   return  <div className={css({ background: '#111' })} />;
};                                                           /* → */ };

import { Circle } from './panda/jsx';                        /* → */ import { Circle } from './panda/jsx';
                                                             /* → */
function App(){                                              /* → */ function App(){
  return  <Circle _hover={{ position: '[absolute]' }} />;    /* → */   return  <Circle _hover={{ position: 'absolute' }} />;
}                                                            /* → */ }
```

## Resources

* [Rule source](/plugin/src/rules/no-escape-hatch.ts)
* [Test source](/tests/no-escape-hatch.test.ts)
