[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-escape-hatch

Prohibit the use of escape hatch syntax in the code.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
import { css } from './panda/css';

const styles = css({ marginLeft: '[4px]' });
```
```js

import { css } from './panda/css';

function App(){
  return <div className={css({ background: '[#111]' })} />;
};
```
```js

import { Circle } from './panda/jsx';

function App(){
  return <Circle _hover={{ position: '[absolute]' }} />;
}
```

âœ”ï¸ Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ marginLeft: '4' });
```
```js

import { css } from './panda/css';

function App(){
  return <div className={css({ background: 'red.100' })} />;
};
```
```js

import { Circle } from './panda/jsx';

function App(){
  return <Circle _hover={{ position: 'absolute' }} />;
}
```

## Resources

* [Rule source](/plugin/src/rules/no-escape-hatch.ts)
* [Test source](/tests/no-escape-hatch.test.ts)
