[//]: # 'This file is generated by eslint-docgen. Do not edit it directly.'

# no-escape-hatch

Prohibit the use of escape hatch syntax in the code.

📋 This rule is enabled in `plugin:@pandacss/all`.

🔧 The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems)
can automatically fix some of the problems reported by this rule.

## Rule details

❌ Examples of **incorrect** code:

```js
import { css } from './panda/css'
import { styled, Circle } from './panda/jsx'

const layout = css({
  display: 'grid',
  gridTemplateColumns: `
[
  [full-start]
    minmax(16px, 1fr)
      [breakout-start]
        minmax(0, 16px)
          [content-start]
            minmax(min-content, 1024px)
          [content-end]
        minmax(0, 16px)
      [breakout-end]
    minmax(16px, 1fr)
  [full-end]
]
`,
})
import { css } from './panda/css'
import { styled, Circle } from './panda/jsx'

;<Circle
  gridTemplateColumns={`
[
  [full-start]
    minmax(16px, 1fr)
      [breakout-start]
        minmax(0, 16px)
          [content-start]
            minmax(min-content, 1024px)
          [content-end]
        minmax(0, 16px)
      [breakout-end]
    minmax(16px, 1fr)
  [full-end]
]
`}
/>
```

✔️ Examples of **correct** code:

```js
import { css } from './panda/css'
import { styled, Circle } from './panda/jsx'

const layout = css({
  display: 'grid',
  gridTemplateColumns: `[full-start]
    minmax(16px, 1fr)
      [breakout-start]
        minmax(0, 16px)
          [content-start]
            minmax(min-content, 1024px)
          [content-end]
        minmax(0, 16px)
      [breakout-end]
    minmax(16px, 1fr)
  [full-end]`,
})
import { css } from './panda/css'
import { styled, Circle } from './panda/jsx'

;<Circle
  gridTemplateColumns={`[full-start]
    minmax(16px, 1fr)
      [breakout-start]
        minmax(0, 16px)
          [content-start]
            minmax(min-content, 1024px)
          [content-end]
        minmax(0, 16px)
      [breakout-end]
    minmax(16px, 1fr)
  [full-end]`}
/>
```

🔧 Examples of code **fixed** by this rule:

```js
import { css } from './panda/css';            /* → */ import { css } from './panda/css';
import { styled, Circle } from './panda/jsx'; /* → */ import { styled, Circle } from './panda/jsx';
                                              /* → */
const layout = css({                          /* → */ const layout = css({
    display: "grid",                          /* → */     display: "grid",
    gridTemplateColumns: `                    /* → */     gridTemplateColumns: `[full-start]
[                                             /* → */     minmax(16px, 1fr)
  [full-start]                                /* → */       [breakout-start]
    minmax(16px, 1fr)                         /* → */         minmax(0, 16px)
      [breakout-start]                        /* → */           [content-start]
        minmax(0, 16px)                       /* → */             minmax(min-content, 1024px)
          [content-start]                     /* → */           [content-end]
            minmax(min-content, 1024px)       /* → */         minmax(0, 16px)
          [content-end]                       /* → */       [breakout-end]
        minmax(0, 16px)                       /* → */     minmax(16px, 1fr)
      [breakout-end]                          /* → */   [full-end]`,
    minmax(16px, 1fr)                         /* → */   });
  [full-end]                                  /* → */   ;
]                                             /* → */
`,                                            /* → */
  });                                         /* → */
  ;                                           /* → */

import { css } from './panda/css';            /* → */ import { css } from './panda/css';
import { styled, Circle } from './panda/jsx'; /* → */ import { styled, Circle } from './panda/jsx';
                                              /* → */
<Circle gridTemplateColumns={`                /* → */ <Circle gridTemplateColumns={`[full-start]
[                                             /* → */     minmax(16px, 1fr)
  [full-start]                                /* → */       [breakout-start]
    minmax(16px, 1fr)                         /* → */         minmax(0, 16px)
      [breakout-start]                        /* → */           [content-start]
        minmax(0, 16px)                       /* → */             minmax(min-content, 1024px)
          [content-start]                     /* → */           [content-end]
            minmax(min-content, 1024px)       /* → */         minmax(0, 16px)
          [content-end]                       /* → */       [breakout-end]
        minmax(0, 16px)                       /* → */     minmax(16px, 1fr)
      [breakout-end]                          /* → */   [full-end]`} />
    minmax(16px, 1fr)                         /* → */
  [full-end]                                  /* → */
]                                             /* → */
`} />                                         /* → */
```

## Resources

- [Rule source](/plugin/src/rules/no-escape-hatch.ts)
- [Test source](/tests/no-escape-hatch.test.ts)
