[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-escape-hatch

Prohibit the use of escape hatch syntax in the code.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/all`.

ğŸ”§ The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## Rule details

âŒ Examples of **incorrect** code for a file named `invalid.tsx`:
```js
import { css } from './panda/css'
import { Circle } from './panda/jsx'

const styles = css({ marginLeft: "[4px]" });

import { css } from './panda/css'
import { Circle } from './panda/jsx'

const layout = css({
    display: "grid",
    gridTemplateColumns: `
[
  [full-start]
    minmax(16px, 1fr)
      [breakout-start]
        minmax(0, 16px)
          [content-start]
            minmax(min-content, 1024px)
          [content-end]
        minmax(0, 16px)
      [breakout-end]
    minmax(16px, 1fr)
  [full-end]
]
`,
  });
  ;

import { css } from './panda/css'
import { Circle } from './panda/jsx'

<div className={css({ background: "[#111]" })} />;

import { css } from './panda/css'
import { Circle } from './panda/jsx'

<Circle _hover={{ position: "[absolute]" }} />;

import { css } from './panda/css'
import { Circle } from './panda/jsx'

<Circle gridTemplateColumns={`
[
  [full-start]
    minmax(16px, 1fr)
      [breakout-start]
        minmax(0, 16px)
          [content-start]
            minmax(min-content, 1024px)
          [content-end]
        minmax(0, 16px)
      [breakout-end]
    minmax(16px, 1fr)
  [full-end]
]
`} />
```

âœ”ï¸ Examples of **correct** code for a file named `valid.tsx`:
```js
import { css } from './panda/css'
import { Circle } from './panda/jsx'

const styles = css({ marginLeft: "4" });

import { css } from './panda/css'
import { Circle } from './panda/jsx'

const layout = css({
      display: "grid",
      gridTemplateColumns: `[full-start]
    minmax(16px, 1fr)
      [breakout-start]
        minmax(0, 16px)
          [content-start]
            minmax(min-content, 1024px)
          [content-end]
        minmax(0, 16px)
      [breakout-end]
    minmax(16px, 1fr)
  [full-end]`,
    });
    ;

import { css } from './panda/css'
import { Circle } from './panda/jsx'

<div className={css({ background: "red.100" })} />;

import { css } from './panda/css'
import { Circle } from './panda/jsx'

<Circle _hover={{ position: "absolute" }} />;

import { css } from './panda/css'
import { Circle } from './panda/jsx'

<Circle gridTemplateColumns={`[full-start]
    minmax(16px, 1fr)
      [breakout-start]
        minmax(0, 16px)
          [content-start]
            minmax(min-content, 1024px)
          [content-end]
        minmax(0, 16px)
      [breakout-end]
    minmax(16px, 1fr)
  [full-end]`} />
```

ğŸ”§ Examples of code **fixed** by this rule for a file named `invalid.tsx`:
```js
import { css } from './panda/css'                  /* â†’ */ import { css } from './panda/css'
import { Circle } from './panda/jsx'               /* â†’ */ import { Circle } from './panda/jsx'
                                                   /* â†’ */
const styles = css({ marginLeft: "[4px]" });       /* â†’ */ const styles = css({ marginLeft: "4px" });

import { css } from './panda/css'                  /* â†’ */ import { css } from './panda/css'
import { Circle } from './panda/jsx'               /* â†’ */ import { Circle } from './panda/jsx'
                                                   /* â†’ */
const layout = css({                               /* â†’ */ const layout = css({
    display: "grid",                               /* â†’ */     display: "grid",
    gridTemplateColumns: `                         /* â†’ */     gridTemplateColumns: `[full-start]
[                                                  /* â†’ */     minmax(16px, 1fr)
  [full-start]                                     /* â†’ */       [breakout-start]
    minmax(16px, 1fr)                              /* â†’ */         minmax(0, 16px)
      [breakout-start]                             /* â†’ */           [content-start]
        minmax(0, 16px)                            /* â†’ */             minmax(min-content, 1024px)
          [content-start]                          /* â†’ */           [content-end]
            minmax(min-content, 1024px)            /* â†’ */         minmax(0, 16px)
          [content-end]                            /* â†’ */       [breakout-end]
        minmax(0, 16px)                            /* â†’ */     minmax(16px, 1fr)
      [breakout-end]                               /* â†’ */   [full-end]`,
    minmax(16px, 1fr)                              /* â†’ */   });
  [full-end]                                       /* â†’ */   ;
]                                                  /* â†’ */
`,                                                 /* â†’ */
  });                                              /* â†’ */
  ;                                                /* â†’ */

import { css } from './panda/css'                  /* â†’ */ import { css } from './panda/css'
import { Circle } from './panda/jsx'               /* â†’ */ import { Circle } from './panda/jsx'
                                                   /* â†’ */
<div className={css({ background: "[#111]" })} />; /* â†’ */ <div className={css({ background: "#111" })} />;

import { css } from './panda/css'                  /* â†’ */ import { css } from './panda/css'
import { Circle } from './panda/jsx'               /* â†’ */ import { Circle } from './panda/jsx'
                                                   /* â†’ */
<Circle _hover={{ position: "[absolute]" }} />;    /* â†’ */ <Circle _hover={{ position: "absolute" }} />;

import { css } from './panda/css'                  /* â†’ */ import { css } from './panda/css'
import { Circle } from './panda/jsx'               /* â†’ */ import { Circle } from './panda/jsx'
                                                   /* â†’ */
<Circle gridTemplateColumns={`                     /* â†’ */ <Circle gridTemplateColumns={`[full-start]
[                                                  /* â†’ */     minmax(16px, 1fr)
  [full-start]                                     /* â†’ */       [breakout-start]
    minmax(16px, 1fr)                              /* â†’ */         minmax(0, 16px)
      [breakout-start]                             /* â†’ */           [content-start]
        minmax(0, 16px)                            /* â†’ */             minmax(min-content, 1024px)
          [content-start]                          /* â†’ */           [content-end]
            minmax(min-content, 1024px)            /* â†’ */         minmax(0, 16px)
          [content-end]                            /* â†’ */       [breakout-end]
        minmax(0, 16px)                            /* â†’ */     minmax(16px, 1fr)
      [breakout-end]                               /* â†’ */   [full-end]`} />
    minmax(16px, 1fr)                              /* â†’ */
  [full-end]                                       /* â†’ */
]                                                  /* â†’ */
`} />                                              /* â†’ */
```

## Resources

* [Rule source](/plugin/src/rules/no-escape-hatch.ts)
* [Test source](/tests/no-escape-hatch.test.ts)
