[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-escape-hatch

Prohibit the use of escape hatch syntax in the code.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/all`.

ğŸ”§ The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## Rule details

âŒ Examples of **incorrect** code:
```js
import { css } from './panda/css';

const styles = css({ marginLeft: '[4px]' });
;

import { css } from './panda/css';

function App(){
  return  <div className={css({ background: '[#111]' })} />;
}
;

import { Circle } from './panda/jsx';

function App(){
  return  <Circle _hover={{ position: '[absolute]' }} />;
}
```

âœ”ï¸ Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ marginLeft: '4' });
;

import { css } from './panda/css';

function App(){
  return  <div className={css({ background: 'red.100' })} />;
}
;

import { Circle } from './panda/jsx';

function App(){
  return  <Circle _hover={{ position: 'absolute' }} />;
}
```

ğŸ”§ Examples of code **fixed** by this rule:
```js
import { css } from './panda/css';                           /* â†’ */ import { css } from './panda/css';
                                                             /* â†’ */
const styles = css({ marginLeft: '[4px]' });                 /* â†’ */ const styles = css({ marginLeft: '4px' });
;                                                            /* â†’ */ ;

import { css } from './panda/css';                           /* â†’ */ import { css } from './panda/css';
                                                             /* â†’ */
function App(){                                              /* â†’ */ function App(){
  return  <div className={css({ background: '[#111]' })} />; /* â†’ */   return  <div className={css({ background: '#111' })} />;
}                                                            /* â†’ */ }
;                                                            /* â†’ */ ;

import { Circle } from './panda/jsx';                        /* â†’ */ import { Circle } from './panda/jsx';
                                                             /* â†’ */
function App(){                                              /* â†’ */ function App(){
  return  <Circle _hover={{ position: '[absolute]' }} />;    /* â†’ */   return  <Circle _hover={{ position: 'absolute' }} />;
}                                                            /* â†’ */ }
```

## Resources

* [Rule source](/plugin/src/rules/no-escape-hatch.ts)
* [Test source](/tests/no-escape-hatch.test.ts)
