[//]: # 'This file is generated by eslint-docgen. Do not edit it directly.'

# no-hardcoded-color

Enforce the exclusive use of design tokens as values for colors within the codebase.

üìã This rule is enabled in `plugin:@pandacss/all`.

## Rule details

‚ùå Examples of **incorrect** code for a file named `invalid.tsx`:

```js
import { css } from './panda/css'
import { Circle } from './panda/jsx'

const styles = css({ color: 'skyblue' })

import { css } from './panda/css'
import { Circle } from './panda/jsx'
;<div className={css({ background: '#111' })} />

import { css } from './panda/css'
import { Circle } from './panda/jsx'
;<Circle _hover={{ borderColor: 'rgb(1, 1, 1)' }} />

import { css } from './panda/css'
import { Circle } from './panda/jsx'
;<Circle _hover={{ bg: 'hsl(0deg, 0%, 7%)' }} />

import { css } from './panda/css'
import { Circle } from './panda/jsx'
;<Circle _hover={{ bgColor: 'rgba(17, 17, 17, 1)' }} />
```

‚úîÔ∏è Examples of **correct** code for a file named `valid.tsx`:

```js
import { css } from './panda/css'
import { Circle } from './panda/jsx'

const styles = css({ color: 'red.100' })

import { css } from './panda/css'
import { Circle } from './panda/jsx'
;<div className={css({ background: 'green.300' })} />

import { css } from './panda/css'
import { Circle } from './panda/jsx'
;<Circle _hover={{ borderColor: 'gray.100' }} />

import { css } from './panda/css'
import { Circle } from './panda/jsx'
;<Circle _hover={{ bg: 'gray.100' }} />

import { css } from './panda/css'
import { Circle } from './panda/jsx'
;<Circle _hover={{ bgColor: 'gray.100' }} />
```

## Resources

- [Rule source](/plugin/src/rules/no-hardcoded-color.ts)
- [Test source](/tests/no-hardcoded-color.test.ts)
