[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-important

Disallow usage of important keyword. Prioroitize specificity for a maintainable and predictable styling structure.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
import { css } from './panda/css';

const styles = css({ marginLeft: '4px!' });
```
```js

import { css } from './panda/css';

function App(){
  return <div className={css({ background: '#111 !important' })} />;
};
```
```js

import { Circle } from './panda/jsx';

function App(){
  return <Circle _hover={{ position: '[absolute!]' }} />;
}
```

âœ”ï¸ Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ marginLeft: '4' });
```
```js

import { css } from './panda/css';

function App(){
  return <div className={css({ background: 'red.100' })} />;
};
```
```js

import { Circle } from './panda/jsx';

function App(){
  return <Circle _hover={{ position: 'absolute' }} />;
}
```

## Resources

* [Rule source](/plugin/src/rules/no-important.ts)
* [Test source](/plugin/tests/no-important.test.ts)
