[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-invalid-nesting

Warn against invalid nesting. Nested styles must contain the `&` character.

📋 This rule is enabled in `plugin:@pandacss/all`.

📋 This rule is enabled in `plugin:@pandacss/recommended`.

## Rule details

❌ Examples of **incorrect** code:
```js
import { css } from './panda/css';

const styles = css({
  ':hover': { marginLeft: '4px' },
});
```
```js

import { css } from './panda/css';

function App() {
  return (
    <div
      className={css({
        '.dark ': { background: 'red.100' },
      })}
    />
  );
};
```
```js

import { Circle } from './panda/jsx';

function App() {
  return (
    <Circle
      css={{
        '[data-focus]': { position: 'absolute' },
      }}
    />
  );
}
```

✔️ Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({
  '&:hover': { marginLeft: '4px' },
});
```
```js

import { css } from './panda/css';

function App() {
  return (
    <div
      className={css({
        '.dark &': { background: 'red.100' },
      })}
    />
  );
};
```
```js

import { Circle } from './panda/jsx';

function App() {
  return (
    <Circle
      css={{
        '&[data-focus]': { position: 'absolute' },
      }}
    />
  );
}
```

## Resources

* [Rule source](/plugin/src/rules/no-invalid-nesting.ts)
* [Test source](/plugin/tests/no-invalid-nesting.test.ts)
