[//]: # 'This file is generated by eslint-docgen. Do not edit it directly.'

# no-invalid-token-paths

Disallow the use of invalid token paths within token function syntax.

üìã This rule is enabled in `plugin:@pandacss/all`.

üìã This rule is enabled in `plugin:@pandacss/recommended`.

## Rule details

‚ùå Examples of **incorrect** code for a file named `invalid.tsx`:

```js
import { css } from './panda/css'
import { Circle } from './panda/jsx'

const styles = css({ bg: 'token(colors.red0.300) 50%' })

import { css } from './panda/css'
import { Circle } from './panda/jsx'

const styles = css({ bg: `token(colors.red0.300) 50%` })

import { css } from './panda/css'
import { Circle } from './panda/jsx'

const styles = css({ bg: 'token(colors.red.3004, red) 50%' })

import { css } from './panda/css'
import { Circle } from './panda/jsx'
;<div className={css({ marginX: '{sizess.2} {sizes.200}' })} />

import { css } from './panda/css'
import { Circle } from './panda/jsx'
;<Circle bg="token(colorss.green.400) 40%" />
```

‚úîÔ∏è Examples of **correct** code for a file named `valid.tsx`:

```js
import { css } from './panda/css'
import { Circle } from './panda/jsx'

const styles = css({ bg: 'token(colors.red.300) 50%' })

import { css } from './panda/css'
import { Circle } from './panda/jsx'

const styles = css({ bg: 'token(colors.red.300, red) 50%' })

import { css } from './panda/css'
import { Circle } from './panda/jsx'
;<div className={css({ marginX: '{sizes.4} {sizes.2}' })} />

import { css } from './panda/css'
import { Circle } from './panda/jsx'
;<Circle bg="token(colors.green.400) 40%" />

import { css } from './panda/css'
import { Circle } from './panda/jsx'
;<Circle bg={`token(colors.green.400) 40%`} />
```

## Resources

- [Rule source](/plugin/src/rules/no-invalid-token-paths.ts)
- [Test source](/tests/no-invalid-token-paths.test.ts)
