[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-margin-properties

Discourage using margin properties for spacing; prefer defining spacing in parent elements with `flex` or `grid` using the `gap` property for a more resilient layout. Margins make components less reusable in other contexts.

üìã This rule is enabled in `plugin:@pandacss/all`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
import { css } from './panda/css';

const styles = css({ marginLeft: '4' });
```
```js

import { css } from './panda/css';

function App(){
  return <div className={css({ margin: '3' })} />;
};
```
```js

import { Circle } from './panda/jsx';

function App(){
  return <Circle marginX="2" />;
}
```

‚úîÔ∏è Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ display: 'flex', gap: '4' });
```
```js

import { grid } from './panda/css';

function App(){
  return <div className={grid({ gap: '3' })} />;
};
```
```js

import { Flex } from './panda/jsx';

function App(){
  return <Flex gap="2" />;
}
```

## Resources

* [Rule source](/plugin/src/rules/no-margin-properties.ts)
* [Test source](/plugin/tests/no-margin-properties.test.ts)
