[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-property-renaming

Ensure user does not rename a property for a pattern or style prop. It doesn't get tracked.

üìã This rule is enabled in `plugin:@pandacss/all`.

üìã This rule is enabled in `plugin:@pandacss/recommended`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
import { css } from './panda/css';

function Text({ variant }){
  return <p className={css({ textStyle: variant })} />;
}
;

import { css } from './panda/css';

function Text(props){
  return <p className={css({ textStyle: props.variant })} />;
}
;

import { Circle } from './panda/jsx';

function CustomCircle(props){
  const { circleSize = '3' } = props
  return (
    <Circle
      size={circleSize}
    />
  )
}
;

import { Circle } from './panda/jsx';

function CustomCircle(props){
  return (
    <Circle
      size={props.circleSize}
    />
  )
}
```

‚úîÔ∏è Examples of **correct** code:
```js
import { css } from './panda/css';

function Text({ textStyle }){
  return <p className={css({ textStyle })} />;
}
;

import { css } from './panda/css';

function Text(props){
  return <p className={css({ textStyle: props.textStyle })} />;
}
;

import { Circle } from './panda/jsx';

function CustomCircle(props){
  const { size = '3' } = props
  return (
    <Circle
      size={size}
    />
  )
}
;

import { Circle } from './panda/jsx';

function CustomCircle(props){
  return (
    <Circle
      size={props.size}
    />
  )
}
```

## Resources

* [Rule source](/plugin/src/rules/no-property-renaming.ts)
* [Test source](/tests/no-property-renaming.test.ts)
