[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-property-renaming

Ensure user does not rename a property for a pattern or style prop. It doesn't get tracked.

📋 This rule is enabled in `plugin:@pandacss/all`.

📋 This rule is enabled in `plugin:@pandacss/recommended`.

## Rule details

❌ Examples of **incorrect** code for a file named `invalid.tsx`:
```js
import { Circle } from './panda/jsx'

const CustomCircle = (props: Props) => {
  const { circleSize = '3' } = props
  return (
    <Circle
      size={circleSize}
    />
  )
}
;

import { Circle } from './panda/jsx'

const CustomCircle = (props: Props) => {
  return (
    <Circle
      size={props.circleSize}
    />
  )
}
```

✔️ Examples of **correct** code for a file named `valid.tsx`:
```js
import { Circle } from './panda/jsx'

const CustomCircle = (props: Props) => {
  const { size = '3' } = props
  return (
    <Circle
      size={size}
    />
  )
}
;

import { Circle } from './panda/jsx'

const CustomCircle = (props: Props) => {
  return (
    <Circle
      size={props.size}
    />
  )
}
```

## Resources

* [Rule source](/plugin/src/rules/no-property-renaming.ts)
* [Test source](/tests/no-property-renaming.test.ts)
