[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-property-renaming

Ensure that properties for patterns or style props are not renamed, as it prevents proper tracking.

📋 This rule is enabled in `plugin:@pandacss/all`.

📋 This rule is enabled in `plugin:@pandacss/recommended`.

## Rule details

❌ Examples of **incorrect** code:
```js
import { css } from './panda/css';

function Text({ variant }){
  return <p className={css({ textStyle: variant })} />;
};
```
```js

import { css } from './panda/css';

function Text(props){
  return <p className={css({ textStyle: props.variant })} />;
}
```

✔️ Examples of **correct** code:
```js
import { css } from './panda/css';

function Text({ textStyle }){
  return <p className={css({ textStyle })} />;
};
```
```js

import { css } from './panda/css';

function Text(props){
  return <p className={css({ textStyle: props.textStyle })} />;
};
```
```js

import { Circle } from './panda/jsx';

function CustomCircle(props){
  const { size = '3' } = props
  return (
    <Circle
      size={size}
    />
  )
};
```
```js

import { Circle } from './panda/jsx';

function CustomCircle(props){
  return (
    <Circle
      size={props.size}
    />
  )
}
```

## Resources

* [Rule source](/plugin/src/rules/no-property-renaming.ts)
* [Test source](/plugin/tests/no-property-renaming.test.ts)
