[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-physical-properties

Encourage the use of logical properties over physical properties to foster a responsive and adaptable user interface.

üìã This rule is enabled in `plugin:@pandacss/all`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
import { css } from './panda/css';

const styles = css({ left: '0' });
```
```js

import { css } from './panda/css';

function App(){
  return <div className={css({ marginLeft: '4' })} />;
};
```
```js

import { Circle } from './panda/jsx';

function App(){
  return <Circle _hover={{  borderBottom: 'solid 1px' }} />;
}
```

‚úîÔ∏è Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ insetInlineStart: '0' });
```
```js

import { css } from './panda/css';

function App(){
  return <div className={css({ marginInlineStart: '4' })} />;
};
```
```js

import { Circle } from './panda/jsx';

function App(){
  return <Circle _hover={{  borderBlockEnd: 'solid 1px' }} />;
}
```

## Resources

* [Rule source](/plugin/src/rules/no-physical-properties.ts)
* [Test source](/plugin/tests/no-physical-properties.test.ts)
