[//]: # 'This file is generated by eslint-docgen. Do not edit it directly.'

# no-shorthand-prop

Discourage the use of shorthand properties and promote the preference for longhand CSS properties in the codebase.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/all`.

ğŸ”§ The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems)
can automatically fix some of the problems reported by this rule.

## Rule details

âŒ Examples of **incorrect** code for a file named `invalid.tsx`:

```js
import { css } from './panda/css'
import { Circle } from './panda/jsx'

const styles = css({ ml: '4' })

import { css } from './panda/css'
import { Circle } from './panda/jsx'
;<div className={css({ bg: 'red.100' })} />

import { css } from './panda/css'
import { Circle } from './panda/jsx'
;<Circle _hover={{ pos: 'absolute' }} />
```

âœ”ï¸ Examples of **correct** code for a file named `valid.tsx`:

```js
import { css } from './panda/css'
import { Circle } from './panda/jsx'

const styles = css({ marginLeft: '4' })

import { css } from './panda/css'
import { Circle } from './panda/jsx'
;<div className={css({ background: 'red.100' })} />

import { css } from './panda/css'
import { Circle } from './panda/jsx'
;<Circle _hover={{ position: 'absolute' }} />
```

ğŸ”§ Examples of code **fixed** by this rule for a file named `invalid.tsx`:

```js
import { css } from './panda/css'           /* â†’ */ import { css } from './panda/css'
import { Circle } from './panda/jsx'        /* â†’ */ import { Circle } from './panda/jsx'
                                            /* â†’ */
const styles = css({ ml: "4" });            /* â†’ */ const styles = css({ marginLeft: "4" });

import { css } from './panda/css'           /* â†’ */ import { css } from './panda/css'
import { Circle } from './panda/jsx'        /* â†’ */ import { Circle } from './panda/jsx'
                                            /* â†’ */
<div className={css({ bg: "red.100" })} />; /* â†’ */ <div className={css({ background: "red.100" })} />;

import { css } from './panda/css'           /* â†’ */ import { css } from './panda/css'
import { Circle } from './panda/jsx'        /* â†’ */ import { Circle } from './panda/jsx'
                                            /* â†’ */
<Circle _hover={{ pos: "absolute" }} />     /* â†’ */ <Circle _hover={{ position: "absolute" }} />
```

## Resources

- [Rule source](/plugin/src/rules/no-shorthand-prop.ts)
- [Test source](/tests/no-shorthand-prop.test.ts)
