[//]: # 'This file is generated by eslint-docgen. Do not edit it directly.'

# no-shorthand-prop

Discourage the use of shorthand properties and promote the preference for longhand CSS properties in the codebase.

📋 This rule is enabled in `plugin:@pandacss/all`.

🔧 The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems)
can automatically fix some of the problems reported by this rule.

## Rule details

❌ Examples of **incorrect** code for a file named `invalid.tsx`:

```js
import { css } from './panda/css'
import { Circle } from './panda/jsx'

const styles = css({ ml: '4' })

import { css } from './panda/css'
import { Circle } from './panda/jsx'
;<div className={css({ bg: 'red.100' })} />

import { css } from './panda/css'
import { Circle } from './panda/jsx'
;<Circle _hover={{ pos: 'absolute' }} />
```

✔️ Examples of **correct** code for a file named `valid.tsx`:

```js
import { css } from './panda/css'
import { Circle } from './panda/jsx'

const styles = css({ marginLeft: '4' })

import { css } from './panda/css'
import { Circle } from './panda/jsx'
;<div className={css({ background: 'red.100' })} />

import { css } from './panda/css'
import { Circle } from './panda/jsx'
;<Circle _hover={{ position: 'absolute' }} />
```

🔧 Examples of code **fixed** by this rule for a file named `invalid.tsx`:

```js
import { css } from './panda/css'           /* → */ import { css } from './panda/css'
import { Circle } from './panda/jsx'        /* → */ import { Circle } from './panda/jsx'
                                            /* → */
const styles = css({ ml: "4" });            /* → */ const styles = css({ marginLeft: "4" });

import { css } from './panda/css'           /* → */ import { css } from './panda/css'
import { Circle } from './panda/jsx'        /* → */ import { Circle } from './panda/jsx'
                                            /* → */
<div className={css({ bg: "red.100" })} />; /* → */ <div className={css({ background: "red.100" })} />;

import { css } from './panda/css'           /* → */ import { css } from './panda/css'
import { Circle } from './panda/jsx'        /* → */ import { Circle } from './panda/jsx'
                                            /* → */
<Circle _hover={{ pos: "absolute" }} />     /* → */ <Circle _hover={{ position: "absolute" }} />
```

## Resources

- [Rule source](/plugin/src/rules/no-shorthand-prop.ts)
- [Test source](/tests/no-shorthand-prop.test.ts)
