[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-shorthand-prop

Discourage the use of shorthand properties and promote the preference for longhand CSS properties in the codebase.

üìã This rule is enabled in `plugin:@pandacss/all`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
import { css } from './panda/css';

const styles = css({ ml: '4' });
```
```js

import { css } from './panda/css';

function App(){
  return  <div className={css({ bg: 'red.100' })} />;
};
```
```js

import { Circle } from './panda/jsx';

function App(){
  return  <Circle _hover={{  pos: 'absolute' }} />;
}
```

‚úîÔ∏è Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ marginLeft: '4' });
```
```js

import { css } from './panda/css';

function App(){
  return  <div className={css({ background: 'red.100' })} />;
};
```
```js

import { Circle } from './panda/jsx';

function App(){
  return  <Circle _hover={{  position: 'absolute' }} />;
}
```

## Resources

* [Rule source](/plugin/src/rules/no-shorthand-prop.ts)
* [Test source](/tests/no-shorthand-prop.test.ts)
