[//]: # 'This file is generated by eslint-docgen. Do not edit it directly.'

# no-unsafe-token-fn-usage

Prevent users from using the token function in situations where they could simply use the raw design token.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/all`.

ğŸ”§ The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems)
can automatically fix some of the problems reported by this rule.

## Rule details

âŒ Examples of **incorrect** code:

```js
import { css } from './panda/css'
import { token as tk } from './panda/tokens'
import { Circle } from './panda/jsx'

const styles = css({ bg: tk('colors.red.300') })

import { css } from './panda/css'
import { token as tk } from './panda/tokens'
import { Circle } from './panda/jsx'

const styles = css({ bg: tk(`colors.red.300`) })

import { css } from './panda/css'
import { token as tk } from './panda/tokens'
import { Circle } from './panda/jsx'

;<Circle bg={tk('colors.red.300')} />
```

âœ”ï¸ Examples of **correct** code:

```js
import { css } from './panda/css'
import { token as tk } from './panda/tokens'
import { Circle } from './panda/jsx'

const styles = css({ bg: 'token(colors.red.300) 50%' })
```

ğŸ”§ Examples of code **fixed** by this rule:

```js
import { css } from './panda/css';                /* â†’ */ import { css } from './panda/css';
import { token as tk } from './panda/tokens'      /* â†’ */ import { token as tk } from './panda/tokens'
import { Circle } from './panda/jsx';             /* â†’ */ import { Circle } from './panda/jsx';
                                                  /* â†’ */
const styles = css({ bg: tk("colors.red.300") }); /* â†’ */ const styles = css({ bg: "red.300" });

import { css } from './panda/css';                /* â†’ */ import { css } from './panda/css';
import { token as tk } from './panda/tokens'      /* â†’ */ import { token as tk } from './panda/tokens'
import { Circle } from './panda/jsx';             /* â†’ */ import { Circle } from './panda/jsx';
                                                  /* â†’ */
const styles = css({ bg: tk(`colors.red.300`) }); /* â†’ */ const styles = css({ bg: "red.300" });

import { css } from './panda/css';                /* â†’ */ import { css } from './panda/css';
import { token as tk } from './panda/tokens'      /* â†’ */ import { token as tk } from './panda/tokens'
import { Circle } from './panda/jsx';             /* â†’ */ import { Circle } from './panda/jsx';
                                                  /* â†’ */
<Circle bg={tk("colors.red.300")} />              /* â†’ */ <Circle bg={"red.300"} />
```

## Resources

- [Rule source](/plugin/src/rules/no-unsafe-token-fn-usage.ts)
- [Test source](/tests/no-unsafe-token-fn-usage.test.ts)
