[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-unsafe-token-fn-usage

Prevent users from using the token function in situations where they could simply use the raw design token.

📋 This rule is enabled in `plugin:@pandacss/all`.

🔧 The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## Rule details

❌ Examples of **incorrect** code:
```js
import { token } from './panda/tokens';
import { css } from './panda/css';

const styles = css({ bg: token('colors.red.300') });

import { token } from './panda/tokens';
import { css } from './panda/css';

function App(){
  return  <div className={css({ bg: 'token(colors.red.300)' })} />;
};

import { Circle } from './panda/jsx';

function App(){
  return  <Circle margin='{sizes.4}' />;
}
```

✔️ Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ bg: 'token(colors.red.300) 50%' });

import { css } from './panda/css';
import { token } from './panda/tokens';

function App(){
  return  <div style={{ color: token('colors.red.50') }} />;
};

import { Circle } from './panda/jsx';

function App(){
  return  <Circle _hover={{  border: 'solid 1px {colors.blue.400}' }} />;
}
```

🔧 Examples of code **fixed** by this rule:
```js
import { token } from './panda/tokens';                             /* → */ import { token } from './panda/tokens';
import { css } from './panda/css';                                  /* → */ import { css } from './panda/css';
                                                                    /* → */
const styles = css({ bg: token('colors.red.300') });                /* → */ const styles = css({ bg: 'red.300' });

import { token } from './panda/tokens';                             /* → */ import { token } from './panda/tokens';
import { css } from './panda/css';                                  /* → */ import { css } from './panda/css';
                                                                    /* → */
function App(){                                                     /* → */ function App(){
  return  <div className={css({ bg: 'token(colors.red.300)' })} />; /* → */   return  <div className={css({ bg: 'red.300' })} />;
};                                                                  /* → */ };

import { Circle } from './panda/jsx';                               /* → */ import { Circle } from './panda/jsx';
                                                                    /* → */
function App(){                                                     /* → */ function App(){
  return  <Circle margin='{sizes.4}' />;                            /* → */   return  <Circle margin='4' />;
}                                                                   /* → */ }
```

## Resources

* [Rule source](/plugin/src/rules/no-unsafe-token-fn-usage.ts)
* [Test source](/tests/no-unsafe-token-fn-usage.test.ts)
