[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-unsafe-token-fn-usage

Prevent users from using the token function in situations where they could simply use the raw design token.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/all`.

ğŸ”§ The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## Rule details

âŒ Examples of **incorrect** code:
```js
import { token } from './panda/tokens';
import { css } from './panda/css';

const styles = css({ bg: token('colors.red.300') });
;

import { token } from './panda/tokens';
import { css } from './panda/css';

function App(){
  return  <div className={css({ bg: 'token(colors.red.300)' })} />;
}
;

import { Circle } from './panda/jsx';

function App(){
  return  <Circle margin='{sizes.4}' />;
}
```

âœ”ï¸ Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ bg: 'token(colors.red.300) 50%' });
;

import { css } from './panda/css';
import { token } from './panda/tokens';

function App(){
  return  <div style={{ color: token('colors.red.50') }} />;
}
;

import { Circle } from './panda/jsx';

function App(){
  return  <Circle _hover={{  border: 'solid 1px {colors.blue.400}' }} />;
}
```

ğŸ”§ Examples of code **fixed** by this rule:
```js
import { token } from './panda/tokens';                             /* â†’ */ import { token } from './panda/tokens';
import { css } from './panda/css';                                  /* â†’ */ import { css } from './panda/css';
                                                                    /* â†’ */
const styles = css({ bg: token('colors.red.300') });                /* â†’ */ const styles = css({ bg: 'red.300' });
;                                                                   /* â†’ */ ;

import { token } from './panda/tokens';                             /* â†’ */ import { token } from './panda/tokens';
import { css } from './panda/css';                                  /* â†’ */ import { css } from './panda/css';
                                                                    /* â†’ */
function App(){                                                     /* â†’ */ function App(){
  return  <div className={css({ bg: 'token(colors.red.300)' })} />; /* â†’ */   return  <div className={css({ bg: 'red.300' })} />;
}                                                                   /* â†’ */ }
;                                                                   /* â†’ */ ;

import { Circle } from './panda/jsx';                               /* â†’ */ import { Circle } from './panda/jsx';
                                                                    /* â†’ */
function App(){                                                     /* â†’ */ function App(){
  return  <Circle margin='{sizes.4}' />;                            /* â†’ */   return  <Circle margin='4' />;
}                                                                   /* â†’ */ }
```

## Resources

* [Rule source](/plugin/src/rules/no-unsafe-token-fn-usage.ts)
* [Test source](/tests/no-unsafe-token-fn-usage.test.ts)
