[//]: # 'This file is generated by eslint-docgen. Do not edit it directly.'

# no-unsafe-token-fn-usage

Prevent users from using the token function in situations where they could simply use the raw design token.

📋 This rule is enabled in `plugin:@pandacss/all`.

🔧 The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems)
can automatically fix some of the problems reported by this rule.

## Rule details

❌ Examples of **incorrect** code:

```js
import { css } from './panda/css'
import { token as tk } from './panda/tokens'
import { Circle } from './panda/jsx'

const styles = css({ bg: tk('colors.red.300') })

import { css } from './panda/css'
import { token as tk } from './panda/tokens'
import { Circle } from './panda/jsx'

const styles = css({ bg: tk(`colors.red.300`) })

import { css } from './panda/css'
import { token as tk } from './panda/tokens'
import { Circle } from './panda/jsx'

;<Circle bg={tk('colors.red.300')} />
```

✔️ Examples of **correct** code:

```js
import { css } from './panda/css'
import { token as tk } from './panda/tokens'
import { Circle } from './panda/jsx'

const styles = css({ bg: 'token(colors.red.300) 50%' })
```

🔧 Examples of code **fixed** by this rule:

```js
import { css } from './panda/css';                /* → */ import { css } from './panda/css';
import { token as tk } from './panda/tokens'      /* → */ import { token as tk } from './panda/tokens'
import { Circle } from './panda/jsx';             /* → */ import { Circle } from './panda/jsx';
                                                  /* → */
const styles = css({ bg: tk("colors.red.300") }); /* → */ const styles = css({ bg: "red.300" });

import { css } from './panda/css';                /* → */ import { css } from './panda/css';
import { token as tk } from './panda/tokens'      /* → */ import { token as tk } from './panda/tokens'
import { Circle } from './panda/jsx';             /* → */ import { Circle } from './panda/jsx';
                                                  /* → */
const styles = css({ bg: tk(`colors.red.300`) }); /* → */ const styles = css({ bg: "red.300" });

import { css } from './panda/css';                /* → */ import { css } from './panda/css';
import { token as tk } from './panda/tokens'      /* → */ import { token as tk } from './panda/tokens'
import { Circle } from './panda/jsx';             /* → */ import { Circle } from './panda/jsx';
                                                  /* → */
<Circle bg={tk("colors.red.300")} />              /* → */ <Circle bg={"red.300"} />
```

## Resources

- [Rule source](/plugin/src/rules/no-unsafe-token-fn-usage.ts)
- [Test source](/tests/no-unsafe-token-fn-usage.test.ts)
