[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-unsafe-token-fn-usage

Prevent users from using the token function in situations where they could simply use the raw design token.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/all`.

ğŸ”§ The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## Rule details

âŒ Examples of **incorrect** code for a file named `invalid.tsx`:
```js
import { css } from './panda/css';
import { Circle } from './panda/jsx';
import { tokens as tk } from './panda/tokens'

const styles = css({ bg: tk("colors.red.300") });

import { css } from './panda/css';
import { Circle } from './panda/jsx';
import { tokens as tk } from './panda/tokens'

const styles = css({ bg: "token(colors.red.300)" });

import { css } from './panda/css';
import { Circle } from './panda/jsx';
import { tokens as tk } from './panda/tokens'

<Circle _hover={{ color: "{colors.blue.300}" }} />
```

âœ”ï¸ Examples of **correct** code for a file named `valid.tsx`:
```js
import { css } from './panda/css';
import { Circle } from './panda/jsx';
import { tokens as tk } from './panda/tokens'

const styles = css({ bg: "token(colors.red.300) 50%" });

import { css } from './panda/css';
import { Circle } from './panda/jsx';
import { tokens as tk } from './panda/tokens'

<div className={css({ border: "solid {borderWidths.1} blue" })} />
```

ğŸ”§ Examples of code **fixed** by this rule for a file named `invalid.tsx`:
```js
import { css } from './panda/css';                   /* â†’ */ import { css } from './panda/css';
import { Circle } from './panda/jsx';                /* â†’ */ import { Circle } from './panda/jsx';
import { tokens as tk } from './panda/tokens'        /* â†’ */ import { tokens as tk } from './panda/tokens'
                                                     /* â†’ */
const styles = css({ bg: tk("colors.red.300") });    /* â†’ */ const styles = css({ bg: "red.300" });

import { css } from './panda/css';                   /* â†’ */ import { css } from './panda/css';
import { Circle } from './panda/jsx';                /* â†’ */ import { Circle } from './panda/jsx';
import { tokens as tk } from './panda/tokens'        /* â†’ */ import { tokens as tk } from './panda/tokens'
                                                     /* â†’ */
const styles = css({ bg: "token(colors.red.300)" }); /* â†’ */ const styles = css({ bg: "red.300" });

import { css } from './panda/css';                   /* â†’ */ import { css } from './panda/css';
import { Circle } from './panda/jsx';                /* â†’ */ import { Circle } from './panda/jsx';
import { tokens as tk } from './panda/tokens'        /* â†’ */ import { tokens as tk } from './panda/tokens'
                                                     /* â†’ */
<Circle _hover={{ color: "{colors.blue.300}" }} />   /* â†’ */ <Circle _hover={{ color: "blue.300" }} />
```

## Resources

* [Rule source](/plugin/src/rules/no-unsafe-token-fn-usage.ts)
* [Test source](/test/no-unsafe-token-fn-usage.test.ts)
