[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# prefer-composite-properties

Encourage the use of composite properties instead of atomic properties in the codebase.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
import { css } from './panda/css';

const styles = css({ rowGap: '4', columnGap: '4' });
```
```js

import { css } from './panda/css';

function App(){
  return <div className={css({ bgColor: 'red.100' })} />;
};
```
```js

import { Circle } from './panda/jsx';

function App(){
  return <Circle _hover={{  borderTopStyle: 'solid', borderTopWidth: '1px', borderTopColor: 'blue' }} />;
}
```

âœ”ï¸ Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ gap: '4' });
```
```js

import { css } from './panda/css';

function App(){
  return <div className={css({ background: 'red.100' })} />;
};
```
```js

import { Circle } from './panda/jsx';

function App(){
  return <Circle _hover={{  borderTop: 'solid 1px blue' }} />;
}
```

## Resources

* [Rule source](/plugin/src/rules/prefer-composite-properties.ts)
* [Test source](/plugin/tests/prefer-composite-properties.test.ts)
