[//]: # 'This file is generated by eslint-docgen. Do not edit it directly.'

# prefer-atomic-properties

Encourage the use of atomic properties instead of composite shorthand properties in the codebase.

üìã This rule is enabled in `plugin:@pandacss/all`.

## Rule details

‚ùå Examples of **incorrect** code for a file named `invalid.tsx`:

```js
import { css } from './panda/css'
import { Circle } from './panda/jsx'

const styles = css({ gap: '4' })

import { css } from './panda/css'
import { Circle } from './panda/jsx'

;<div className={css({ background: 'red' })} />

import { css } from './panda/css'
import { Circle } from './panda/jsx'

;<Circle _hover={{ borderTop: 'solid 1px blue' }} />
```

‚úîÔ∏è Examples of **correct** code for a file named `valid.tsx`:

```js
import { css } from './panda/css'
import { Circle } from './panda/jsx'

const styles = css({ rowGap: '4', columnGap: '4' })

import { css } from './panda/css'
import { Circle } from './panda/jsx'

;<div className={css({ backgroundColor: 'red' })} />

import { css } from './panda/css'
import { Circle } from './panda/jsx'

;<Circle _hover={{ borderTopStyle: 'solid', borderTopWidth: '1px', borderTopColor: 'blue' }} />
```

## Resources

- [Rule source](/plugin/src/rules/prefer-atomic-properties.ts)
- [Test source](/tests/prefer-atomic-properties.test.ts)
