[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# prefer-atomic-properties

Encourage the use of atomic properties instead of composite shorthand properties in the codebase.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
import { css } from './panda/css';

const styles = css({ gap: '4' });

import { css } from './panda/css';

function App(){
  return  <div className={css({ background: 'red.100' })} />;
};

import { Circle } from './panda/jsx';

function App(){
  return  <Circle _hover={{  borderTop: 'solid 1px blue' }} />;
}
```

âœ”ï¸ Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ rowGap: '4', columnGap: '4' });

import { css } from './panda/css';

function App(){
  return  <div className={css({ backgroundColor: 'red.100' })} />;
};

import { Circle } from './panda/jsx';

function App(){
  return  <Circle _hover={{  borderTopStyle: 'solid', borderTopWidth: '1px', borderTopColor: 'blue' }} />;
}
```

## Resources

* [Rule source](/plugin/src/rules/prefer-atomic-properties.ts)
* [Test source](/tests/prefer-atomic-properties.test.ts)
