[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# prefer-shorthand-properties

Discourage the use of longhand properties and promote the preference for shorthand properties in the codebase.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/all`.

## Rule details

âŒ Examples of **incorrect** code:
```js
import { css } from './panda/css';

const styles = css({ marginLeft: '4' });
```
```js

import { css } from './panda/css';

function App(){
  return <div className={css({ background: 'red.100' })} />;
};
```
```js

import { Circle } from './panda/jsx';

function App(){
  return <Circle _hover={{  position: 'absolute' }} />;
}
```

âœ”ï¸ Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ ml: '4' });
```
```js

import { css } from './panda/css';

function App(){
  return <div className={css({ bg: 'red.100' })} />;
};
```
```js

import { Circle } from './panda/jsx';

function App(){
  return <Circle _hover={{  pos: 'absolute' }} />;
}
```

## Resources

* [Rule source](/plugin/src/rules/prefer-shorthand-properties.ts)
* [Test source](/tests/prefer-shorthand-properties.test.ts)
