[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# prefer-unified-property-style

Discourage mixing atomic and composite forms of the same property in a style declaration. Atomic styles give more consistent results.

üìã This rule is enabled in `plugin:@pandacss/all`.

## Rule details

‚ùå Examples of **incorrect** code:
```js
import { css } from './panda/css';

const styles = css({ margin:"2", marginLeft: "5" });
```
```js

import { Circle } from './panda/jsx';

function App(){
  return <Circle border="solid 1px" borderColor="gray.800" />;
}
```

‚úîÔ∏è Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ marginTop: "2", marginRight: "2", marginBottom: "2", marginLeft: "5" });
```
```js

import { Circle } from './panda/jsx';

function App(){
  return <Circle borderStyle="solid" borderColor="gray.900" borderWidth="1px" />;
}
```

## Resources

* [Rule source](/plugin/src/rules/prefer-unified-property-style.ts)
* [Test source](/plugin/tests/prefer-unified-property-style.test.ts)
