[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# prefer-unified-property-style

Discourage against mixing atomic and composite forms of the same property in a style declaration. Atomic styles give more consistent results

ğŸ“‹ This rule is enabled in `plugin:@pandacss/all`.

ğŸ“‹ This rule is enabled in `plugin:@pandacss/recommended`.

## Rule details

âŒ Examples of **incorrect** code:
```js
import { css } from './panda/css';

const styles = css({ margin:"2", marginLeft: "5" });
```
```js

import { Circle } from './panda/jsx';

function App(){
  return <Circle border="solid 1px" borderColor="gray.800" />;
}
```

âœ”ï¸ Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ marginTop: "2", marginRight: "2", marginBottom: "2", marginLeft: "5" });
```
```js

import { Circle } from './panda/jsx';

function App(){
  return <Circle borderStyle="solid" borderColor="gray.900" borderWidth="1px" />;
}
```

## Resources

* [Rule source](/plugin/src/rules/prefer-unified-property-style.ts)
* [Test source](/tests/prefer-unified-property-style.test.ts)
