[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# prefer-longhand-properties

Discourage the use of shorthand properties and promote the preference for longhand properties in the codebase.

📋 This rule is enabled in `plugin:@pandacss/all`.

## Rule details

❌ Examples of **incorrect** code:
```js
import { css } from './panda/css';

const styles = css({ ml: '4' });
```
```js

import { css } from './panda/css';

function App(){
  return <div className={css({ bg: 'red.100' })} />;
};
```
```js

import { Circle } from './panda/jsx';

function App(){
  return <Circle _hover={{  pos: 'absolute' }} />;
}
```

✔️ Examples of **correct** code:
```js
import { css } from './panda/css';

const styles = css({ marginLeft: '4' });
```
```js

import { css } from './panda/css';

function App(){
  return <div className={css({ background: 'red.100' })} />;
};
```
```js

import { Circle } from './panda/jsx';

function App(){
  return <Circle _hover={{  position: 'absolute' }} />;
}
```

## Resources

* [Rule source](/plugin/src/rules/prefer-longhand-properties.ts)
* [Test source](/plugin/tests/prefer-longhand-properties.test.ts)
