diff --git a/src/write-docs-from-tests.js b/src/write-docs-from-tests.js
index b409d5df46b7bc6ec8b15b1c94f2569d87857adc..b7fda61f7c766c1015f8e3f28e9bc1daf9b808be 100644
--- a/src/write-docs-from-tests.js
+++ b/src/write-docs-from-tests.js
@@ -83,7 +83,7 @@ async function writeDocsFromTests( name, rule, tests, testerConfig, done ) {
 					throw new Error( errors.map( formatter.error ).join( '\n' ) );
 				}
 
-				done();
+				return;
 			}
 		);
 	} );