# @pandacss/no-debug

âš ï¸ This rule _warns_ in the following configs: ğŸŒ `plugin:@pandacss/all`, âœ… `plugin:@pandacss/recommended`.

ğŸ”§ This rule is automatically fixable by the
[`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

<!-- end auto-generated rule header -->

Disallow the inclusion of the debug attribute when shipping code to the production environment.

```json
{
  "plugins": ["@pandacss"],
  "rules": {
    "@pandacss/no-debug": "warn" // or error
  }
}
```

```js
// âŒ bad
const styles = css({ debug: true })

// âœ… good
const styles = css({ ... })
```
