# @pandacss/no-dynamic-styling

âš ï¸ This rule _warns_ in the following configs: ğŸŒ `plugin:@pandacss/all`, âœ… `plugin:@pandacss/recommended`.

<!-- end auto-generated rule header -->

Ensure user doesn't use dynamic styling at any point. Prefer to use static styles, leverage css variables or recipes for
known dynamic styles.

```json
{
  "plugins": ["@pandacss"],
  "rules": {
    "@pandacss/no-dynamic-styling": "warn" // or error
  }
}
```

```js
// âŒ bad
const color = 'red'
const styles = css({ background: color })

// âœ… good
const styles = css({ background: 'red' })
```
