# @pandacss/no-hardcoded-color

âš ï¸ This rule _warns_ in the ğŸŒ `plugin:@pandacss/all` config.

<!-- end auto-generated rule header -->

Enforce the exclusive use of design tokens as values for colors within the codebase.

```json
{
  "plugins": ["@pandacss"],
  "rules": {
    "@pandacss/no-hardcoded-color": "warn" // or error
  }
}
```

```js
// âŒ bad
const styles = css({ background: '#111' })

// âœ… good
const styles = css({ background: 'gray.100' })
```
