# @pandacss/no-invalid-token-paths

ğŸ’¼ This rule is enabled in the following configs: ğŸŒ `plugin:@pandacss/all`, âœ… `plugin:@pandacss/recommended`.

<!-- end auto-generated rule header -->

Disallow the use of invalid token paths within token function syntax.

```json
{
  "plugins": ["@pandacss"],
  "rules": {
    "@pandacss/no-invalid-token-paths": "warn" // or error
  }
}
```

```js
// âŒ bad                                      ^
const styles = css({ border: 'solid 1px {colorss.gray.50}' })

// âœ… good
const styles = css({ border: 'solid 1px {colors.gray.50}' })
```
