# @pandacss/prefer-atomic-properties

âš ï¸ This rule _warns_ in the ğŸŒ `plugin:@pandacss/all` config.

<!-- end auto-generated rule header -->

Encourage the use of atomic properties instead of composite shorthand properties in the codebase.

```json
{
  "plugins": ["@pandacss"],
  "rules": {
    "@pandacss/prefer-atomic-properties": "warn" // or error
  }
}
```

```js
// âŒ bad
const styles = css({ gap: '4' })

// âœ… good
const styles = css({ columngGap: '4', rowGap: '4' })
```
